central/policy/datastore/datastore_impl.go

```diff
@@ -411,16 +387,62 @@ func (ds *datastoreImpl) importPolicy(ctx context.Context, policy *storage.Polic
 	return result
 }
 
-func (ds *datastoreImpl) policyNameIsNotUnique(policyNameIDMap map[string]string, name string) bool {
-	for n := range policyNameIDMap {
+func (ds *datastoreImpl) validateUniqueNameAndID(ctx context.Context, policy *storage.Policy, result *v1.ImportPolicyResponse,
+	overwrite bool, policyNameToPolicyMap map[string]*storage.Policy) ([]*v1.ImportPolicyError, error) {
+	var importErrors []*v1.ImportPolicyError
+
+	if policy.GetId() != "" {
+		existingPolicy, exists, err := ds.storage.Get(ctx, policy.GetId())
+		if err != nil {
+			result.Errors = getImportErrorsFromError(err)
+			return importErrors, err
+		}
+		if exists {
+			// New policy cannot have the same id as default policies (aka system policies) and it cannot be overwritten, so always return an error
+			if existingPolicy.GetIsDefault() {
+				importErrors = append(importErrors,
+					duplicateNameImportErr(fmt.Sprintf("system policy with id '%q' already exists, unable to import policy", policy.GetId()), policiesPkg.ErrImportDuplicateSystemPolicyID, existingPolicy.GetName()))
```

Comment:
Maybe just making this Errf? so don't have to sprintf inside?

Reply:
I guess I meant to change the function to be
func duplicateNameImportErr(errType string, duplicateName string, template string, args ...interface{}) {
  then you could fmt.Sprintf(template, args...) just once in the function
}